File: backend/internal/domain/scene.go
package domain

import "time"

type ScenarioStatus string

const (
	ScenarioStatusDraft     ScenarioStatus = "draft"
	ScenarioStatusPending   ScenarioStatus = "pending"
	ScenarioStatusApproved  ScenarioStatus = "approved"
	ScenarioStatusRejected  ScenarioStatus = "rejected"
	ScenarioStatusPublished ScenarioStatus = "published"
	ScenarioStatusArchived  ScenarioStatus = "archived"
)

type Scenario struct {
	ID              string         `json:"id"`
	Name            string         `json:"name"`
	Code            string         `json:"code"`
	CoverImage      *string        `json:"coverImage,omitempty"`
	CareerPositionID *string       `json:"careerPositionId,omitempty"`
	IndustryIDs     []string       `json:"industryIds,omitempty"`
	IndustryNames   []string       `json:"industryNames,omitempty"`
	ProfessionIDs   []string       `json:"professionIds,omitempty"`
	ProfessionNames []string       `json:"professionNames,omitempty"`
	BatchID         *string        `json:"batchId,omitempty"`
	Difficulty      int            `json:"difficulty"`
	Version         string         `json:"version"`
	Status          ScenarioStatus `json:"status"`
	Background      *string        `json:"background,omitempty"`
	DeliveryGoal    *string        `json:"deliveryGoal,omitempty"`
	CreatorID       string         `json:"creatorId"`
	CoBuilderIDs    []string       `json:"coBuilderIds,omitempty"`
	TenantID        *string        `json:"tenantId,omitempty"`
	CreatedAt       time.Time      `json:"createdAt"`
	UpdatedAt       time.Time      `json:"updatedAt"`
	PublishTime     *time.Time     `json:"publishTime,omitempty"`
	ViewCount       int            `json:"viewCount"`
}

type ScenarioTask struct {
	ID                  string              `json:"id"`
	ScenarioID          string              `json:"scenarioId"`
	Name                string              `json:"name"`
	Code                string              `json:"code"`
	SortOrder           int                 `json:"sortOrder"`
	Description         *string             `json:"description,omitempty"`
	DetailedDescription *string             `json:"detailedDescription,omitempty"`
	DescriptionPdf      *string             `json:"descriptionPdf,omitempty"`
	EstimatedHours      float64             `json:"estimatedHours"`
	TaskType            string              `json:"taskType"`
	Difficulty          int                 `json:"difficulty"`
	Background          *string             `json:"background,omitempty"`
	DependencyIDs       []string            `json:"dependencyIds,omitempty"`
	IsReferenced        bool                `json:"isReferenced"`
	SourceScenarioID    *string             `json:"sourceScenarioId,omitempty"`
	KnowledgePointIDs   []string            `json:"knowledgePointIds,omitempty"`
	AbilityPointIDs     []string            `json:"abilityPointIds,omitempty"`
	ResourceIDs         []string            `json:"resourceIds,omitempty"`
	EvalData            JSONMap             `json:"evalData,omitempty"`
	TenantID            *string             `json:"tenantId,omitempty"`
}

type TaskEvaluationConfig struct {
	ID            string    `json:"id"`
	TaskID        string    `json:"taskId"`
	MethodKey     string    `json:"methodKey"`
	Weight        float64   `json:"weight"`
	EvalObjects   JSONMap   `json:"evalObjects,omitempty"`
	EvalSubjects  JSONSlice `json:"evalSubjects,omitempty"`
	EvalResources JSONMap   `json:"evalResources,omitempty"`
}

type TaskEvalPoint struct {
	ID                string   `json:"id"`
	ConfigID          string   `json:"configId"`
	Name              string   `json:"name"`
	Description       *string  `json:"description,omitempty"`
	Weight            float64  `json:"weight"`
	MaxScore          float64  `json:"maxScore"`
	ScoringMethod     string   `json:"scoringMethod"`
	GradeMapping      JSONMap  `json:"gradeMapping,omitempty"`
	SubType           *string  `json:"subType,omitempty"`
	KnowledgePointIDs []string `json:"knowledgePointIds,omitempty"`
	AbilityPointIDs   []string `json:"abilityPointIds,omitempty"`
	SortOrder         int      `json:"sortOrder"`
}

type TaskReviewStep struct {
	ID          string  `json:"id"`
	ConfigID    string  `json:"configId"`
	Label       string  `json:"label"`
	Description *string `json:"description,omitempty"`
	Enabled     bool    `json:"enabled"`
	SubjectType *string `json:"subjectType,omitempty"`
	Weight      float64 `json:"weight"`
	SortOrder   int     `json:"sortOrder"`
}

type TaskResource struct {
	ID                string    `json:"id"`
	Name              string    `json:"name"`
	Type              string    `json:"type"`
	URL               *string   `json:"url,omitempty"`
	Description       *string   `json:"description,omitempty"`
	Thumbnail         *string   `json:"thumbnail,omitempty"`
	Size              *string   `json:"size,omitempty"`
	KnowledgePointIDs []string  `json:"knowledgePointIds,omitempty"`
	ExtraData         JSONMap   `json:"extraData,omitempty"`
	UploadedBy        *string   `json:"uploadedBy,omitempty"`
	UploadedAt        time.Time `json:"uploadedAt"`
}

type TaskResourceBinding struct {
	ID         string `json:"id"`
	TaskID     string `json:"taskId"`
	ResourceID string `json:"resourceId"`
}

type TaskKnowledgeBinding struct {
	ID               string `json:"id"`
	TaskID           string `json:"taskId"`
	KnowledgePointID string `json:"knowledgePointId"`
}

type TaskAbilityBinding struct {
	ID             string `json:"id"`
	TaskID         string `json:"taskId"`
	AbilityPointID string `json:"abilityPointId"`
}

type ScenarioWeightConfig struct {
	ID         string  `json:"id"`
	ScenarioID string  `json:"scenarioId"`
	TaskID     string  `json:"taskId"`
	Weight     float64 `json:"weight"`
}

type ScenarioGradeMapping struct {
	ID          string  `json:"id"`
	ScenarioID  string  `json:"scenarioId"`
	TaskID      *string `json:"taskId,omitempty"`
	Level       string  `json:"level"`
	MinScore    float64 `json:"minScore"`
	MaxScore    float64 `json:"maxScore"`
	Description *string `json:"description,omitempty"`
	Color       *string `json:"color,omitempty"`
}

// SceneBatchStatus removed — defined in status.go

type SceneBatch struct {
	ID            string           `json:"id"`
	Name          string           `json:"name"`
	Code          *string          `json:"code,omitempty"`
	OrgNodeID     *string          `json:"orgNodeId,omitempty"`
	MajorID       *string          `json:"majorId,omitempty"`
	MajorName     *string          `json:"majorName,omitempty"`
	WorkflowID    *string          `json:"workflowId,omitempty"`
	Status        SceneBatchStatus `json:"status"`
	ScenarioCount int              `json:"scenarioCount"`
	CreatedAt     time.Time        `json:"createdAt"`
	UpdatedAt     time.Time        `json:"updatedAt"`
}

File: packages/shared-types/src/scene.ts
export interface Scenario {
  id: string
  name: string
  code: string
  coverImage?: string
  careerPositionId?: string
  industryIds?: string[]
  industryNames?: string[]
  professionIds?: string[]
  professionNames?: string[]
  batchId?: string
  difficulty: number
  version: string
  viewCount?: number
  status: "draft" | "pending" | "approved" | "rejected" | "published" | "archived"
  background?: string
  deliveryGoal?: string
  creatorId: string
  coBuilderIds: string[]
  createdAt: string
  updatedAt: string
  publishTime?: string
}

export interface ScenarioTask {
  id: string
  scenarioId: string
  name: string
  code: string
  sortOrder: number
  description?: string
  detailedDescription?: string
  descriptionPdf?: string
  estimatedHours: number
  taskType: "assessment" | "training"
  difficulty: number
  background?: string
  dependencyIds: string[]
  isReferenced: boolean
  sourceScenarioId?: string
  knowledgePointIds: string[]
  abilityPointIds: string[]
  resourceIds: string[]
  evalData?: Record<string, any>
}

export interface TaskDeliverable {
  id: string
  taskId: string
  type: string
  name: string
  description?: string
  evaluationPoints?: Record<string, any>
  sortOrder: number
}

export interface TaskResource {
  id: string
  name: string
  type: string
  url?: string
  description?: string
  thumbnail?: string
  size?: string
  knowledgePointIds?: string[]
  extraData?: Record<string, any>
  uploadedBy?: string
  uploadedAt: string
}

export interface TaskResourceBinding {
  id: string
  taskId: string
  resourceId: string
}

export interface TaskKnowledgeBinding {
  id: string
  taskId: string
  knowledgePointId: string
}

export interface TaskAbilityBinding {
  id: string
  taskId: string
  abilityPointId: string
}

export interface RubricTemplate {
  id: string
  tenantId: string
  name: string
  mode: "rubric" | "score_rule"
  types?: string[]
  description?: string
  data: Record<string, any>
  createdAt: string
  updatedAt: string
}

export interface TaskEvaluationMethod {
  id: string
  taskId: string
  methodKey: string
  weight: number
  evalObject: string
  scoreType?: string
  evalSubjects: Record<string, any>[]
  rubricTemplateId?: string
  resourceConfig: Record<string, any>
  evalPoints: TaskEvalPoint[]
  reviewSteps: TaskReviewStep[]
}

export interface TaskEvalPoint {
  id: string
  configId: string
  name: string
  description?: string
  subType?: string
  types?: string[]
  weight: number
  scoringMethod: string
  gradeMapping?: Record<string, any>[]
  knowledgePointIds?: string[]
  abilityPointIds?: string[]
  sortOrder: number
}

export interface TaskReviewStep {
  id: string
  configId: string
  label: string
  description?: string
  enabled: boolean
  subjectType?: string
  weight: number
  sortOrder: number
}

export interface ScenarioWeightConfig {
  id: string
  scenarioId: string
  taskId: string
  weight: number
}

export interface ScenarioGradeMapping {
  id: string
  scenarioId: string
  taskId?: string
  level: string
  minScore: number
  maxScore: number
  description?: string
  color?: string
}

export interface SceneArchive {
  id: string
  scenarioId: string
  version: string
  snapshotData: Record<string, any>
  archivedAt: string
}

export interface SceneBatch {
  id: string
  tenantId?: string
  name: string
  code?: string
  orgNodeId?: string
  majorId?: string
  majorName?: string // Deprecated
  workflowId?: string
  status: "open" | "closed"
  scenarioCount?: number
  createdAt: string
  updatedAt: string
}
